assets/js/components/Dashboard/NewPaMessage/SelectStationsPage.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useMemo, useState } from "react";
+import { Button, Container, Form } from "react-bootstrap";
+import fp from "lodash/fp";
+import { useScreenplayContext } from "Hooks/useScreenplayContext";
+import type { Place } from "Models/place";
+import RouteColumn from "./RouteColumn";
+import { GREEN_LINE_ROUTES, SILVER_LINE_ROUTES } from "Constants/constants";
+import StationGroupCheckbox from "./StationGroupCheckbox";
+import SelectedStationsSummary from "./SelectedStationsSummary";
+import {
+  GL_CENTRAL_SUBWAY,
+  GL_D_BRANCH,
+  GL_E_BRANCH,
+  GLX,
+  ORANGE_NORTH,
+  ORANGE_SOUTH,
+  RED_ASHMONT_BRANCH,
+  RED_BRAINTREE_BRANCH,
+  RED_TRUNK,
+} from "./StationGroups";
+
+const usePlacesWithPaEss = () => {
+  const { places } = useScreenplayContext();
+  return useMemo(
+    () =>
+      places
+        .map((place) => ({
+          ...place,
+          screens: place.screens.filter((screen) => screen.type === "pa_ess"),
+        }))
+        .filter((place: Place) => place.screens.length > 0),
+    [places],
+  );
+};
+
+const BASE_PLACE_ROUTE_TO_ROUTE_IDS: { [key: string]: string[] } = {
+  "Green-B": ["Green-B"],
+  "Green-C": ["Green-C"],
+  "Green-D": ["Green-D"],
+  "Green-E": ["Green-E"],
+  Red: ["Red"],
+  Orange: ["Orange"],
+  Blue: ["Blue"],
+  Mattapan: ["Mattapan"],
+  Silver: SILVER_LINE_ROUTES,
+};
+
+const SelectStationsPage = () => {
+  const [zones, setZones] = useState<string[]>([]);
```

Comment:
**note:** this will need extracted to the parent component so state is maintained across sub-pages of the new pa/ess messsage flow. we can do that later though if you'd prefer

Reply:
Hmm yeah good point. Since the Cancel button reverts all changes back to whatever the initial state was (always empty here but when editing it will hold a list of zones) do we need to continue to keep state here and persist it to the parent after confirming the zones? Or is there a more elegant way to hold onto initial state?

Reply:
I think we can take a list of `defaultZones` from the parent which we use to set a local value and then a callback to save to the parent zones when we save.